### PR TITLE
Feat/add browser based tests

### PR DESCRIPTION
**Pull request by chelseapinka**
# Adding Browser based E2E tests

Added two test cases
 - Ensure that VCs can be issued and then revoked
 - Ensure that VCs will not be issued without the required minimum credentials. 
 - Added named exports for modules that were missing them

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by ThisIsMissEm** [comment]
Try moving the waitFor's before the interactions (timing here is tricky, but the click might be happening before the waitFor's — this is why in other tests we assert on DOM elements being added or removed).

also the fill on line 41 should probably be above / outside of that promise.all block

File: .eslintignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ jest.config.ts
 license-header.js
 rollup.config.js
 tsconfig.json
+e2e/browser/test-app
```
Comment (NSeydoux) [inline .eslintignore:5]: Why add the test app to the unlinted files? Does it prevent conflicts with its own linter config?

File: e2e/browser/test-app/components/appContent/index.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  TESTID_OPENID_PROVIDER_INPUT,
+  TESTID_SESSION_STATUS,
+} from "@inrupt/internal-playwright-testids";
+import { useState, useEffect } from "react";
+import VerifiableCredential from "../verifiableCredentials";
+
+// This is the content of the file uploaded manually at SHARED_FILE_IRI.
+const DEFAULT_ISSUER = "https://login.inrupt.com/";
+const REDIRECT_URL = window.location.href;
+const APP_NAME = "Access Grants browser-based tests app";
```
Comment (NSeydoux) [inline e2e/browser/test-app/components/appContent/index.tsx:41]:
```suggestion
const APP_NAME = "Verifiable Credentials browser-based tests app";
```

File: e2e/browser/test-app/components/appContent/index.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+const AccessGrantContainer = ({
```
Comment (NSeydoux) [inline e2e/browser/test-app/components/appContent/index.tsx:42]: Even if the VCs here happen to be access grants, I would be in favour of gearing the naming towards VC not to emphasis on the data model, but on its spec-compliant container.

File: e2e/browser/test-app/styles/Home.module.css [hunk 4]
```diff
@@ -0,0 +1,129 @@
+.container {
```
Comment (NSeydoux) [inline e2e/browser/test-app/styles/Home.module.css:1]: I think it's fine to strip the test app of all styling

File: e2e/browser/test/e2e.playwright.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  await page.fill("input[data-testid=vcProvider]", vcProvider || "");
+  // Create the resource
+  await Promise.all([
+    page.click("button[data-testid=create-resource]"),
+    page.waitForRequest((request) => request.method() === "POST"),
+    page.waitForResponse((response) => response.status() === 201),
+  ]);
+  await expect(
+    page.innerText("span[data-testid=resource-iri]")
+  ).resolves.toMatch(/https:\/\/.*\.txt/);
```
Comment (NSeydoux) [inline e2e/browser/test/e2e.playwright.ts:44]: I wouldn't bother creating a dedicated resource. It makes sense if testing Access Grant capabilities, i.e. "can I use the issued Access Grant to get access to a resource", but all we want to test here is "can I issue a VC passing some data in and checking that said data made it through". The fact that the VC happens to be an Access Grant is not that important, as the Access Grant features are tested in the dedicated library.

That means this step can be removed, as well as the related components in the test app.

Comment (NSeydoux) [reply e2e/browser/test/e2e.playwright.ts:44]:
```suggestion
```

File: e2e/browser/test/e2e.playwright.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  await Promise.all([
+    page.click("button[data-testid=issue-vc]"),
+    page.waitForRequest((request) => request.method() === "POST"),
+    page.waitForResponse((response) => response.status() === 201),
+  ]);
+  await expect(
+    page.innerText("pre[data-testid=verifiable-credential]")
+  ).resolves.not.toBe("");
+
+  // TODO add extra validation of claims here?
```
Comment (NSeydoux) [inline e2e/browser/test/e2e.playwright.ts:56]: That sounds good, maybe parsing the VC as JSON and checking for some properties?

File: e2e/browser/test/e2e.playwright.ts [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+  // Cleanup the resource
+  await Promise.all([
+    page.click("button[data-testid=delete-resource]"),
+    page.waitForRequest((request) => request.method() === "DELETE"),
+    page.waitForResponse((response) => response.status() === 204),
+  ]);
+  await expect(
+    page.innerText("span[data-testid=resource-iri]")
+  ).resolves.toMatch("");
```
Comment (NSeydoux) [inline e2e/browser/test/e2e.playwright.ts:76]:
```suggestion
```
See above comment about VC vs Access Grants

File: e2e/browser/test/e2e.playwright.ts [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  await Promise.all([
+    page.click("button[data-testid=create-resource]"),
+    page.waitForRequest((request) => request.method() === "POST"),
+    page.waitForResponse((response) => response.status() === 201),
+    // eslint-disable-next-line playwright/no-conditional-in-test
+    page.fill("input[data-testid=vcProvider]", vcProvider || ""),
+  ]);
+  await expect(
+    page.innerText("span[data-testid=resource-iri]")
+  ).resolves.toMatch(/https:\/\/.*\.txt/);
```
Comment (NSeydoux) [inline e2e/browser/test/e2e.playwright.ts:94]:
```suggestion
```
Ditto

File: e2e/browser/test/e2e.playwright.ts [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+  // Cleanup the resource
+  await Promise.all([
+    page.click("button[data-testid=delete-resource]"),
+    page.waitForRequest((request) => request.method() === "DELETE"),
+    page.waitForResponse((response) => response.status() === 204),
+  ]);
+  await expect(
+    page.innerText("span[data-testid=resource-iri]")
+  ).resolves.toMatch("");
```
Comment (NSeydoux) [inline e2e/browser/test/e2e.playwright.ts:116]:
```suggestion
```
Ditto

**Review by NSeydoux — COMMENTED** [review]
Just a couple of superficial comments for the test app, but that looks good!

**Review by ThisIsMissEm — COMMENTED** [review]
I read @NSeydoux's comments and agree, I'll review prop on Monday or Tuesday

**Review by NSeydoux — APPROVED** [review]
All good!

**Pull request merged**